semantic_router/index/qdrant.py

```diff
@@ -0,0 +1,226 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from pydantic.v1 import Field
+
+from semantic_router.index.base import BaseIndex
+
+DEFAULT_COLLECTION_NAME = "semantic-router-collection"
+DEFAULT_UPLOAD_BATCH_SIZE = 100
+SCROLL_SIZE = 1000
+SR_UTTERANCE_PAYLOAD_KEY = "sr_utterance"
+SR_ROUTE_PAYLOAD_KEY = "sr_route"
+
+
+class QdrantIndex(BaseIndex):
+    "The name of the collection to use"
+
+    collection_name: str = Field(
+        default=DEFAULT_COLLECTION_NAME,
+        description=f"The name of the Qdrant collection to use. Defaults to '{DEFAULT_COLLECTION_NAME}'",
+    )
+    location: Optional[str] = Field(
+        default=":memory:",
+        description="If ':memory:' - use an in-memory Qdrant instance. Used as 'url' value otherwise",
+    )
+    url: Optional[str] = Field(
+        default=None,
+        description="Qualified URL of the Qdrant instance. Optional[scheme], host, Optional[port], Optional[prefix]",
+    )
+    port: Optional[int] = Field(
+        default=6333,
+        description="Port of the REST API interface.",
+    )
+    grpc_port: int = Field(
+        default=6334,
+        description="Port of the gRPC interface.",
+    )
+    prefer_grpc: bool = Field(
+        default=None,
+        description="Whether to use gPRC interface whenever possible in methods",
+    )
+    https: Optional[bool] = Field(
+        default=None,
+        description="Whether to use HTTPS(SSL) protocol.",
+    )
+    api_key: Optional[str] = Field(
+        default=None,
+        description="API key for authentication in Qdrant Cloud.",
+    )
+    prefix: Optional[str] = Field(
+        default=None,
+        description="Prefix to the REST URL path. Example: `http://localhost:6333/some/prefix/{qdrant-endpoint}`.",
+    )
+    timeout: Optional[int] = Field(
+        default=None,
+        description="Timeout for REST and gRPC API requests.",
+    )
+    host: Optional[str] = Field(
+        default=None,
+        description="Host name of Qdrant service. If url and host are None, set to 'localhost'.",
+    )
+    path: Optional[str] = Field(
+        default=None,
+        description="Persistence path for Qdrant local",
+    )
+    grpc_options: Optional[Dict[str, Any]] = Field(
+        default=None,
+        description="Options to be passed to the low-level Qdrant GRPC client, if used.",
+    )
+    size: Union[int, None] = Field(
+        default=None,
+        description="Embedding dimensions. Defaults to the embedding length of the configured encoder.",
+    )
+    distance: str = Field(
+        default="Cosine", description="Distance metric to use for similarity search."
+    )
+    collection_options: Optional[Dict[str, Any]] = Field(
```

Comment:
given we are renaming `collection_name` to `index_name`, maybe we can rename this? Maybe `config`? We can apply the same to the `PineconeIndex` where we currently have `ServerlessSpec`

Reply:
`config` would be too generic, don't you think? Since these are options for collection creation specifically.

Reply:
Possibly, what would you recommend? Is it within `collection_options` that I would specify `hnsw_config`, `wal_config`, `optimizers_config`, `shard_number`, `on_disk_payload`, and `quantization_config`?

We want something that can align with other vec dbs or storage options, my thought is that `collection_options` is specific to qdrant which is okay but for people switching between `PineconeIndex` and `QdrantIndex` who are not familiar with both the changing parameter names could be confusing

Reply:
> Is it within collection_options that I would specify hnsw_config, wal_config, optimizers_config, shard_number, on_disk_payload, and quantization_config

Yes.

I can rename it to `config` if that would be beneficial as you mentioned.

Reply:
https://github.com/aurelio-labs/semantic-router/pull/207/commits/622d422c61de50ac710772fce1c89eb1fea61e7e